storage/src/test/java/zipkin2/storage/kafka/KafkaStorageIT.java

```diff
@@ -189,6 +194,25 @@
           return traces.size() == 1
               && traces.get(0).size() == 2; // Trace is found and has two spans
         });
+    await().atMost(10, TimeUnit.SECONDS)
+        .until(() -> {
+          List<List<Span>> traces = new ArrayList<>();
+          try {
+            traces =
+                spanStore.getTraces(QueryRequest.newBuilder()
+                    .endTs(TODAY + 1)
+                    .lookback(Duration.ofMinutes(1).toMillis())
+                    .limit(1)
+                    .build())
+                    .execute();
+          } catch (InvalidStateStoreException e) { // ignoring state issues
+            System.err.println(e.getMessage());
+          } catch (Exception e) {
+            e.printStackTrace();
+          }
+          return traces.size() == 1
+              && traces.get(0).size() == 1; // last trace is returned first
+        });
     await().atMost(5, TimeUnit.SECONDS)
```

Comment:
side question.. why all the waits? I know in some cases you can't get a callback of storage quorum met. in such case you could add a sleep to a command that wraps storage to declutter the IT's main code..

Reply:
Good question, probably copy-pasta heh. I think only first one is needed, will clean up.